index.html

```diff
@@ -2945,6 +3039,118 @@ <h3>
+        </p>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <code>short_name</code> member
+        </h3>
+        <p>
+          The <dfn>short_name</dfn> member of a <a>ShortcutItem</a> is a
+          <a>string</a> that represents a short version of the name of the
+          shortcut. It is intended to be used where there is insufficient space
+          to display the full name of the shortcut.
+        </p>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <code>description</code> member
+        </h3>
+        <p>
+          The <dfn>description</dfn> member of a <a>ShortcutItem</a> is a
+          <a>string</a> that allows the developer to describe the purpose of
+          the shortcut. User agents may choose to expose this information to
+          assistive technology.
+        </p>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <code>url</code> member
+        </h3>
+        <p>
+          The <dfn>url</dfn> member of a <a>ShortcutItem</a> is a <a>URL</a> to
+          which a user agent should navigate when the associated shortcut is
+          activated.
+        </p>
+        <p>
+          The steps for <dfn>processing the <code>url</code> member of a
+          shortcut</dfn> are given by the following algorithm. The algorithm
+          takes a <a>ShortcutItem</a> <var>url</var>, and a <var>manifest
+          URL</var>, which is the <a>URL</a> from which the <var>manifest</var>
+          was fetched. This algorithm will return a <a>URL</a> or
+          <code>undefined</code>.
+        </p>
+        <ol>
+          <li>Let <var>value</var> be <var>url</var>
+          </li>
+          <li>If <a>Type</a>(<var>value</var>) it not <a>String</a>, return
+          <code>undefined</code>.
+          </li>
+          <li>Otherwise, <a>parse</a> <var>value</var> using <var>manifest
+          URL</var> as the base URL and return the result.
+          </li>
+        </ol>
+      </section>
+      <section data-dfn-for="ShortcutItem" data-link-for="ShortcutItem">
+        <h3>
+          <dfn>icons</dfn> member
+        </h3>
+        <p>
+          The <a>icons</a> member of an <a>ShortcutItem</a> member is an
+          <a>array</a> of <a>ImageResource</a>s that can serve as iconic
+          representations of the shortcut in various contexts.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <dfn>Launching a shortcut</dfn>
+        </h3>
+        <p>
+          When <a>ShortcutItem</a> <var>shortcut</var> having
+          <a>WebAppManifest</a> <var>manifest</var> is invoked, run the
+          following steps:
+        </p>
+        <ol>
+          <li>Let <var>url</var> be <var>shortcut.url</var>.
+          </li>
+          <li>Let <var>browsing context</var> be the result of creating a new
```

Comment:
What does "new top-level browsing context" mean here? Would it create a new window in all cases? On desktop platforms, it would rather be common to reuse one of the existing windows (or launch the app with given parameters when it’s closed).

I think these steps are okay if the application isn't open. During runtime, I'd suggest raising an `onshortcutclick` event on a service worker which can then override the default behavior (e.g. by navigating an existing client window or opening a new one).

Reply:
In most cases it would be a fresh launch of the PWA or (potentially) a fresh tab within the context of a PWA (if tabs are supported). There is separate work ongoing with respect to the [launch event](https://github.com/WICG/sw-launch) that feels like it could enable what you are looking for with respect to the service worker.

Reply:
Yep, the `launch` event was [specifically designed with this feature in mind](https://github.com/WICG/sw-launch/blob/master/explainer.md#background).